### PR TITLE
set default value for clusterselector fields

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>

### CODE REVIEW

File: pkg/webhook/cluster/mutating_webhook.go [hunk 1]
```diff
@@ -42,22 +39,137 @@ func (a *ManagedClusterMutatingAdmissionHook) MutatingResource() (schema.GroupVe
...
+			Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+			Message: err.Error(),
+		}
+		return nil, status
+	}
+
+	newClusterSet := clusterSet.DeepCopy()
+
+	if clusterSet.Spec.ClusterSelector.SelectorType == "" {
+		newClusterSet.Spec.ClusterSelector.SelectorType = clusterv1beta1.ExclusiveLabel
```
Comment (qiujian16) [inline pkg/webhook/cluster/mutating_webhook.go:117]: I do not think you need this, it is already set by default

File: pkg/webhook/cluster/mutating_webhook.go [hunk 2]
```diff
@@ -42,22 +39,137 @@ func (a *ManagedClusterMutatingAdmissionHook) MutatingResource() (schema.GroupVe
...
+	}
+
+	if clusterSet.Spec.ClusterSelector.ExclusiveLabel != nil {
```
Comment (qiujian16) [inline pkg/webhook/cluster/mutating_webhook.go:120]: this should be done in validatingwebhook, right?

File: pkg/webhook/cluster/mutating_webhook.go [hunk 3]
```diff
@@ -42,22 +39,137 @@ func (a *ManagedClusterMutatingAdmissionHook) MutatingResource() (schema.GroupVe
...
+			status.Allowed = false
+			status.Result = &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusInternalServerError, Reason: metav1.StatusReasonInternalError,
+				Message: fmt.Sprintf("The spec.clusterSelector.ExclusiveLabel.value must be %q.", clusterSet.Name),
+			}
+			return nil, status
+		}
+	}
+
+	newClusterSet.Spec.ClusterSelector.ExclusiveLabel = &clusterv1beta1.ManagedClusterLabel{
```
Comment (qiujian16) [inline pkg/webhook/cluster/mutating_webhook.go:141]: would you do a comparison here, and noop if this is already set

**Comment by ldpliu** [comment]
close as it's not needed in this release

**Pull request closed without merging**